extensions/2.0/Khronos/KHR_instancing/extensions/2.0/Khronos/KHR_instancing/README.md

```diff
@@ -0,0 +1,61 @@
+# KHR\_instancing
+
+## Khronos 3D Formats Working Group
+
+* John Cooke
+* TODO
+
+## Acknowledgments
+
+* TODO
+
+## Status
+
+Experimental
+
+## Dependencies
+
+Written against the glTF 2.0 spec.
+
+## Overview
+
+This extension is specfically designed to enable GPU instancing which renders many copies of a single mesh at once using a small number of draw calls.  The data consists of 
+a transform and an instance id.  The transforms allow the mesh to be displayed at many different locations with different rotations and scales.  The id can be used to alter
+other parameters based on the run-time inplementation.
+
+## Extending Nodes with per instance attributes
+
+Instancing is defined by adding the `KHR_instancing` extension to any glTF mesh node.  The attributes section contains accessor ids for new TRANSFORM4x3 and ID attribute buffers. 
+For example, the following defines some instancing attributes to a mesh node in the graph. Instancing only applies to mesh nodes.  Applying to nodes rather than meshes allows the 
+same mesh to be attached as normal to nodes while also being instanced.  
+
+```json
+{
+    "nodes": [
+        {
+            "mesh": 0,
+            "name": "teapot",
+            "extensions": {
+                "KHR_instancing": {
+                    "attributes": {
+                        "TRANSFORM4x3": 0,
+                        "ID": 1
```

Comment:
Are these IDs implementation-dependent, with no spec-defined usage? If so, they shouldn't be included here. glTF 2.0 has an established workflow for supporting app-specific attribute semantics - they just start with an underscore.

Otherwise, the spec should be more sound and include at least:
- allowed datatypes (likely scalar uint8/16/32, uint32 is not available on WebGL 1.0);
- uniqueness within a set of instances, within a scene, or within an asset (otherwise they should not be called IDs).

In shaders, GPU instances have built-in IDs. Should this attribute be somehow related to them?